components/bridge/TxModal.tsx

```diff
@@ -89,125 +83,38 @@ const TxModal: React.FC<Props> = ({
 			}
 		}
 		//eslint-disable-next-line
-	}, [etherscanHash, modalState, relayerStatus]);
+	}, [modalAttributes, relayerStatus]);
+
+	useEffect(() => {
+		if (modalOpen && modalAttributes) {
+			if (
+				modalAttributes?.hash &&
+				modalAttributes.hash !== "noTokenSelected" &&
+				modalAttributes.state !== "relayer" &&
+				etherscanLink !== ""
+			) {
+				showDialog({
+					title: modalAttributes.title,
+					message: modalAttributes.text,
+					buttonText: "View on Etherscan",
+					link: etherscanLink,
+					disabled: false,
+					loading: true,
+				});
+			} else {
+				showDialog({
+					title: modalAttributes.title,
+					message: modalAttributes.text,
+					disabled: modalAttributes.state === "withdrawCENNZside",
+					callback: resetModal,
+					loading: modalAttributes.state !== "finished",
+				});
+			}
+		}
+		//eslint-disable-next-line
+	}, [modalOpen, modalAttributes, etherscanLink]);
```

Comment:
cool, can you add another comment above `//eslint-disable-next-line` saying

```
// FIXME: adding '<whatever the dependency name here>' cause infinite loop
```

see `components/bridge/ChainPicker.tsx` for an example